src/parser/job.rs

```diff
@@ -114,10 +115,25 @@ impl std::cmp::PartialEq for Job {
 }
 
 impl Job {
-    pub fn start(self: &mut Self, job_name: &String) {
+    pub fn start(self: &mut Self, job_name: &String, target_process: Option<u32>) {
         println!("log: start {}", job_name);
 
-        for i in 0..self.num_procs {
+        let mut start_index = 0;
+        let mut end_index = self.num_procs;
+        if let Some(nb) = target_process {
+            if nb < self.num_procs {
```

Comment:
Isn't it better to check this in the parsing to handle errors in only one place ? (to send it to the client)

Reply:
well that seams more complicated ans less efficient because when i parse the targetted process, it means I need to check in it if the client argument was filled and if a corresponding job exists  which is done in another place of the code + I need to get the appropriate job

Reply:
Ok, so we need to return a result for start/restart/stop to handle the errors

Reply:
resolved on discord